### PR TITLE
Draft: Feature/GitHub provider

### PR DESCRIPTION
**Pull request by fatihkahveci**
Need to review TODO notes. Also we need to implement Github Enterprise version too.

### CODE REVIEW

File: internal/metrics/cycletime/service.go [hunk 1]
```diff
@@ -139,6 +139,10 @@ func (c *cycleTime) mergeRequestFirstCommit(commits []*models.Commit) *models.Co
 }
 
 func (c cycleTime) mergeRequestFirstComment(comments []*models.Comment) *models.Comment {
+	//TODO: not sure about that null check.
+	if len(comments) == 0 {
+		return &models.Comment{}
```
Comment (c1982) [inline internal/metrics/cycletime/service.go:144]: Yes. Logic is contain null checks a one top layer. if the return is null, the calculate is cancelled, so null return is important.

File: internal/metrics/cycletime/service.go [hunk 2]
```diff
@@ -159,6 +163,10 @@ func (c cycleTime) mergeRequestFirstComment(comments []*models.Comment) *models.
 }
 
 func (c cycleTime) mergeRequestApprovalComment(comments []*models.Comment) *models.Comment {
+	//TODO: not sure about that null check.
+	if len(comments) == 0 {
+		return &models.Comment{}
```
Comment (c1982) [inline internal/metrics/cycletime/service.go:168]: it must be return nil.

File: internal/scm/github/scm.go [hunk 3]
```diff
@@ -1,18 +1,599 @@
...
+
+	return repository, nil
+}
+
+func (s *SCM) ListMergeRequest(state, scope string, createdafterday int) (mergerequests models.MergeRequests, err error) {
+	mergerequests = []models.MergeRequest{}
+	filterdate := time.Now().AddDate(0, 0, -createdafterday)
+	mergedOnly := false
+
+	switch state {
```
Comment (c1982) [inline internal/scm/github/scm.go:147]: This switch case, to better testable and maintainabile may be extract to function.

Comment (fatihkahveci) [reply internal/scm/github/scm.go]: Done

**Review by c1982 — APPROVED** [review]

**Comment by c1982** [comment]
Thanks bro. LGTM! :)

**Pull request merged**